astropy/table/tests/test_serialize.py

```diff
@@ -0,0 +1,347 @@
+
+
+@pytest.mark.skipif('not HAS_YAML')
+@pytest.mark.parametrize('table_cls', (Table, QTable))
+def test_ecsv_mixins_as_one(table_cls):
+    """Test write/read all cols at once and validate intermediate column names"""
+    names = sorted(mixin_cols)
+
+    serialized_names = ['ang',
+                        'cr.x', 'cr.y', 'cr.z',
+                        'dt',
+                        'el.x', 'el.y', 'el.z',
+                        'lat',
+                        'lon',
+                        'nd',
+                        'qdb',
+                        'qdex',
+                        'qmag',
+                        'sc.ra', 'sc.dec',
+                        'scc.x', 'scc.y', 'scc.z',
+                        'scd.ra', 'scd.dec', 'scd.distance',
+                        'scd.obstime',
+                        'sd.d_lon_coslat', 'sd.d_lat', 'sd.d_distance',
+                        'sr.lon', 'sr.lat', 'sr.distance',
+                        'srd.lon', 'srd.lat', 'srd.distance',
+                        'srd.differentials.s.d_lon_coslat',
+                        'srd.differentials.s.d_lat',
+                        'srd.differentials.s.d_distance',
+                        'tm',  # serialize_method is formatted_value
+                        'tm2',  # serialize_method is formatted_value
+                        'tm3.jd1', 'tm3.jd2',    # serialize is jd1_jd2
+                        'tm3.location.x', 'tm3.location.y', 'tm3.location.z',
+                        'x']
+
+    t = table_cls([mixin_cols[name] for name in names], names=names)
+
+    out = StringIO()
+    t.write(out, format="ascii.ecsv")
+    t2 = table_cls.read(out.getvalue(), format='ascii.ecsv')
+
+    assert t.colnames == t2.colnames
+
+    # Read as a ascii.basic table (skip all the ECSV junk)
+    t3 = table_cls.read(out.getvalue(), format='ascii.basic')
+    assert t3.colnames == serialized_names
+
+
+def make_multidim(col, ndim):
+    """Take a col with length=2 and make it N-d by repeating elements.
+
+    For the special case of ndim==1 just return the orignal.
+
+    The output has shape [3] * ndim. By using 3 we can be sure that repeating
+    the two input elements gives an output that is sufficiently unique for
+    the multidim tests.
+    """
+    if ndim > 1:
+        import itertools
+        idxs = [idx for idx, _ in zip(itertools.cycle([0, 1]), range(3 ** ndim))]
+        col = col[idxs].reshape([3] * ndim)
+    return col
+
+
+@pytest.mark.skipif('not HAS_YAML')
+@pytest.mark.parametrize('name_col', list(mixin_cols.items()))
+@pytest.mark.parametrize('table_cls', (Table, QTable))
+@pytest.mark.parametrize('ndim', (1, 2, 3))
+def test_ecsv_mixins_per_column(table_cls, name_col, ndim):
+    """Test write/read one col at a time and do detailed validation.
+    This tests every input column type as 1-d, 2-d and 3-d.
+    """
+    name, col = name_col
+
+    c = make_multidim(np.array([1.0, 2.0]), ndim)
```

Comment:
I think the tests in `test_serialize.py` are mostly the same as the ones moved out of `test_ecsv.py`, with the exception of this one which adds N-d versions of every column type.